crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,541 @@
+                            return false;
+                        }
+                    }
+
+                    ProposalDependency::ProposalCertificate => {
+                        if let HotShotEvent::QuorumProposalRecv(proposal, _) = event {
+                            proposal.data.view_number
+                        } else {
+                            return false;
+                        }
+                    }
+                    ProposalDependency::PayloadAndMetadata => {
+                        if let HotShotEvent::SendPayloadCommitmentAndMetadata(
+                            _payload_commitment,
+                            _metadata,
+                            view,
+                        ) = event
+                        {
+                            *view
+                        } else {
+                            return false;
+                        }
+                    }
+                };
+                event_view == view_number
+            }),
+        )
+    }
+
+    /// Create and store an [`AndDependency`] combining [`EventDependency`]s associated with the
+    /// given view number if it doesn't exist. Also takes in the received `event` to seed a
+    /// dependency as already completed. This allows for the task to receive a proposable event
+    /// without losing the data that it received, as the dependency task would otherwise have no
+    /// ability to receive the event and, thus, would never propose.
+    fn create_dependency_task_if_new(
+        &mut self,
+        view_number: TYPES::Time,
+        event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+        event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+        event: Arc<HotShotEvent<TYPES>>,
+    ) {
+        debug!("Attempting to make dependency task for event {:?}", event);
+        if self.propose_dependencies.get(&view_number).is_some() {
+            debug!("Task already exists");
+            return;
+        }
+
+        let mut proposal_cert_validated_dependency = self.create_event_dependency(
+            ProposalDependency::ProposalCertificate,
+            view_number,
+            event_receiver.clone(),
+        );
+
+        let mut qc_dependency = self.create_event_dependency(
+            ProposalDependency::QC,
+            view_number,
+            event_receiver.clone(),
+        );
+
+        let mut view_sync_dependency = self.create_event_dependency(
+            ProposalDependency::ViewSyncCert,
+            view_number,
+            event_receiver.clone(),
+        );
+
+        let mut timeout_dependency = self.create_event_dependency(
+            ProposalDependency::TimeoutCert,
+            view_number,
+            event_receiver.clone(),
+        );
+
+        let mut payload_commitment_dependency = self.create_event_dependency(
+            ProposalDependency::PayloadAndMetadata,
+            view_number,
+            event_receiver,
+        );
+
+        match event.as_ref() {
```

Comment:
@bfish713 So I split these out to their own dependencies and handled Timeout certs separately as requested (looks like the PR comment was removed).